### PR TITLE
GH-98407: fix `test_kill_issue43884` to not leak child processes 

### PR DESCRIPTION
**Pull request by kumaraditya303**
* Issue: gh-98407

### CODE REVIEW

File: Lib/test/test_asyncio/test_subprocess.py [hunk 1]
```diff
@@ -184,22 +184,24 @@ def test_kill(self):
...
         )
         self.loop.run_until_complete(asyncio.sleep(1))
         if sys.platform == 'win32':
             proc.send_signal(signal.CTRL_BREAK_EVENT)
-        # On windows it is an alias of terminate which sets the return code
-        proc.kill()
+            # On windows it is an alias of terminate which sets the return code
+            proc.kill()
+        else:
+            os.killpg(proc.pid, signal.SIGKILL)
```
Comment (gvanrossum) [inline Lib/test/test_asyncio/test_subprocess.py]: I feel this is not in the spirit of the original repro. I will investigate more.

Comment (kumaraditya303) [reply Lib/test/test_asyncio/test_subprocess.py]: Without the fix this test still reproduces the issue for me. It gives the following traceback without the fix:
```console
/workspaces/cpython/Lib/test/support/__init__.py:738: ResourceWarning: unclosed file <_io.FileIO name=9 mode='rb' closefd=True>
  gc.collect()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/workspaces/cpython/Lib/asyncio/unix_events.py:565: ResourceWarning: unclosed transport <_UnixReadPipeTransport closing fd=9 open>
  _warn(f"unclosed transport {self!r}", ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/workspaces/cpython/Lib/asyncio/base_subprocess.py:125: ResourceWarning: unclosed transport <_UnixSubprocessTransport pid=5685 returncode=-9 stdout=<_UnixReadPipeTransport fd=10 open>>
  _warn(f"unclosed transport {self!r}", ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
Warning -- Unraisable exception
Exception ignored in: <function BaseSubprocessTransport.__del__ at 0x7f87c89533e0>
Traceback (most recent call last):
  File "/workspaces/cpython/Lib/asyncio/base_subprocess.py", line 126, in __del__
    self.close()
  File "/workspaces/cpython/Lib/asyncio/base_subprocess.py", line 104, in close
    proto.pipe.close()
  File "/workspaces/cpython/Lib/asyncio/unix_events.py", line 561, in close
    self._close(None)
  File "/workspaces/cpython/Lib/asyncio/unix_events.py", line 585, in _close
    self._loop.call_soon(self._call_connection_lost, exc)
  File "/workspaces/cpython/Lib/asyncio/base_events.py", line 772, in call_soon
    self._check_closed()
  File "/workspaces/cpython/Lib/asyncio/base_events.py", line 519, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
/workspaces/cpython/Lib/test/support/__init__.py:738: ResourceWarning: unclosed file <_io.FileIO name=10 mode='rb' closefd=True>
  gc.collect()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/workspaces/cpython/Lib/asyncio/unix_events.py:565: ResourceWarning: unclosed transport <_UnixReadPipeTransport closing fd=10 open>
  _warn(f"unclosed transport {self!r}", ResourceWarning, source=self)
ResourceWarning: Enable tracemalloc to get the object allocation traceback
test_asyncio failed (env changed)

== Tests result: SUCCESS ==

1 test altered the execution environment:
    test_asyncio

Total duration: 4.3 sec
Tests result: SUCCESS
```
With fix there is no event loop closed traceback.

**Review by gvanrossum — APPROVED** [review]
I'd like confirmation from the OP (@db3l) that this indeed no longer leaves orphan processes hanging around (for more than 2 seconds :-).

**Comment by db3l** [comment]
@gvanrossum I can confirm that this PR built locally leaves no processes.  I'll also watch for the regular buildbot run post-merge.

**Review by gvanrossum — APPROVED** [review]

**Pull request merged**